dsl_parser/rel_graph.py

```diff
@@ -741,8 +742,8 @@ def _node_instance_id(node_id, ctx):
     return new_node_instance_id
 
 
-def _generate_id():
-    return '%05x' % random.randrange(16 ** 5)
+def _generate_id(id_len=5):
+    return ''.join(choice(digits + ascii_letters) for _ in xrange(id_len))
```

Comment:
Actually, both those options are slower (at least on my machine). 
I've timed them (1M executions), and got: 3.873 secs with my version, 4.004 with the `sample` version, and 4.729 with the double join version.